src/thread_pool.rs

```diff
@@ -1052,6 +1079,13 @@ impl WorkerThread {
         {
             self.promote();
         }
+
+        // Loom can't reasona about the amortized scheduling we are doing, so we
```

Comment:
If I fix the deadlock above and then set `LOOM_MAX_PREEMPTIONS=3`, which [the documentation](https://docs.rs/loom/latest/loom/#large-models) says "is enough to catch most bugs", then the tests complete in a reasonable amount of time even with the heartbeat running!  I have no idea what consequences there might be to changing that setting, though.